src/libraries/System.Security.Permissions/ref/System.Security.Permissions.cs

```diff
@@ -1128,6 +1128,9 @@ public sealed partial class PrincipalPermission : System.Security.IPermission, S
     [System.AttributeUsageAttribute(System.AttributeTargets.Class | System.AttributeTargets.Method, AllowMultiple=true, Inherited=false)]
     public sealed partial class PrincipalPermissionAttribute : System.Security.Permissions.CodeAccessSecurityAttribute
     {
+#if CAS_OBSOLETIONS
+        [System.ObsoleteAttribute("PrincipalPermissionAttribute is not honored by the runtime and must not be used.", true, DiagnosticId = "BCL0002", UrlFormat = "https://aka.ms/dotnet-warnings/{0}")]
```

Comment:
How/where are we tracking the distribution/assignment of BCL#### IDs?

Reply:
I tried cataloging them in a single place as part of https://github.com/dotnet/runtime/pull/37535. The idea was that this file would be included in every project we compile. However, since the IL linker "trim" step doesn't run over every project, it ended up including the internal class in every library we built. :(

Reply:
I added a markdown file to act as a centralized list. Does this help?